### PR TITLE
Report MAX_LOADS_TO_GPU throttling, return load promise and errors, fix node size with orthographic camera

### PR DESCRIPTION
**Pull request by rhuitl**
* Return whether a node has been deferred to the next frame due to MAX_LOADS_TO_GPU

  This is a simple flag that signals to the caller of updatePointClouds() that the method has to be called again to finish loading of nodes.

* Report failures when loading node data and return node load promises to the caller

  Completing the promise chain to resolve a promise when the web worker has posted back the decoded node. Fix house keeping of the number of loading nodes in case of errors (e.g. 404 on a bin or hrc file).

### CODE REVIEW

File: src/loading/binary-loader.ts [hunk 1]
```diff
@@ -93,8 +93,15 @@ export class BinaryLoader {
...
   }
 
-  private parse(node: PointCloudOctreeGeometryNode, buffer: ArrayBuffer): void {
+  private parse(node: PointCloudOctreeGeometryNode, buffer: ArrayBuffer): Promise<void> {
+    return new Promise((resolve) => {
+      this._parse(node, buffer, resolve);
+    });
+  }
+
+  private _parse(node: PointCloudOctreeGeometryNode, buffer: ArrayBuffer, resolve: () => void) {
```
Comment (hccampos) [inline src/loading/binary-loader.ts]: Why not keep this as `parse` and then just do this when loading:

```ts
return Promise.resolve(this.getUrl(this.getNodeUrl(node)))
  .then(url => this.xhrRequest(url, { mode: 'cors' }))
  .then(res => res.arrayBuffer())
  .then(buffer => {
    return new Promise(resolve => this.parse(node, buffer, resolve));
  });
```

File: src/potree.ts [hunk 2]
```diff
@@ -167,13 +175,17 @@ export class Potree implements IPotree {
...
-      unloadedGeometry[i].load();
+      promises.push(unloadedGeometry[i].load());
     }
 
     return {
       visibleNodes: visibleNodes,
       numVisiblePoints: numVisiblePoints,
+      exceededMaxLoadsToGPU: exceededMaxLoadsToGPU,
+      failed: failed,
+      promises: promises
```
Comment (hccampos) [inline src/potree.ts]: I don't think I like this name very much because it doesn't indicate what the promises are for. It would be better to have something like `loadPromises` or `nodeLoadPromisses`.

File: src/potree.ts [hunk 3]
```diff
@@ -225,7 +237,7 @@ export class Potree implements IPotree {
         projectionFactor = halfHeight / (slope * distance);
       } else {
         const orthographic = camera as OrthographicCamera;
-        projectionFactor = radius / orthographic.top;
+        projectionFactor = 2 * halfHeight / (orthographic.top - orthographic.bottom)
```
Comment (hccampos) [inline src/potree.ts]: Could you please make a separate PR for this?

Comment (rhuitl) [reply src/potree.ts]: Sure :) https://github.com/pnext/three-loader/pull/39

File: src/potree.ts (hunk 2 above)
Comment (rhuitl) [reply src/potree.ts]: True, `nodeLoadPromises` and `nodeLoadFailed`? (I'd rename the `failed` one, as well)

Comment (hccampos) [reply src/potree.ts]: Yep, makes sense.

File: src/loading/binary-loader.ts (hunk 1 above)
Comment (rhuitl) [reply src/loading/binary-loader.ts]: Nice :)

**Comment by rhuitl** [comment]
Addressed your comments, thanks!

**Comment by hccampos** [comment]
@rhuitl released just now in v0.0.19
